openfisca-web-api/input-output-data.md

```diff
@@ -1,104 +1,294 @@
-# Input and output data
+# Using the /calculate endpoint
 
-## Values by period
+> All the examples provided here are from the [country package template](https://github.com/openfisca/country-template).
 
-In OpenFisca, the value of a variable is always defined for a specific period.
+In order to run a computation on the Web API, you will need to send information to the API concerning:
+- The situation, meaning describe the entities (e.g. individuals, households) that you want to base your calculations on.
+- The variable you need to compute.
 
-This is encoded in JSON using the following format:
+## Describing the situation
 
-```JSON
-{"salary": {"2015-01": 2000}}
-```
+### Describing entities
+
+The most important rule in describing a situation in OpenFisca is:
+
+Every person has to belong to one of each group entity (e.g. household).
+Every person in a group entity needs a role (e.g. parent)
 
-This format allows you to encode an evolution of the variable over time:
+> For example, if you wish to run a calculation on 2 households. One composed of to adults, the other of one adult and one child.
 
-```JSON
+
+```json
 {
-  "salary": {
-    "2015-01": 2000,
-    "2015-02": 2000,
-    "2015-03": 2100,
-    "2015-04": 2100,
+  "persons": { 
+    "Ricarda": {},
+    "Bob": {},
+    "Bill": {},
+    "Janet": {}
+    },
+  "households": {
+    "household_1": {
+      "parents": [
+        "Ricarda", "Bob"
+      ]
+    },
+    "household_2": {
+      "parents": [
+        "Bill"
+      ],
+      "children":[
+        "Janet"
+      ]
+    }
   }
 }
 ```
 
-When you define the inputs of a simulation, you should in general define the period for which you are setting values (except for variables which cannot change over time, such as a birth date).
+## Adding information to entities
 
-If no period is given, for example `{enfant_a_charge: false}`, your input will **only** be set for the global period of the [scenario](#scenarios).
+To run a precise calculation, you can provide information on each person and group entity.
 
-Example: `{salaire_de_base: 5000}` is equivalent to `{salaire_de_base: {<scenario_period>: 5000}}`, `<scenario_period>` being defined in the [scenario](#scenarios).
 
-## Scenarios
+These information are called input variables and they have three caracteristics:
+ - An entity type (e.g. Salary is an input variable for an Person, housing occupancy status is an input variable for a household)
+ - A date format (e.g. Yearly, monthly ...),
+ - A formula or a default value, that are used for computations when the input variable was not provided.
```

Comment:
A little higher, I can read "you can provide information".
Then "These information"
Then "A formula or a default value, that are used for computations when the input variable was not provided."

This seems to indicate that as an API client, I'm the one to provide these information... but that's not the case. What I provide is values of the variable for specific dates.

Reply:
I don't understand :/

Reply:
If you read the whole paragraph, it seems to say that as a client, I can provide informations about default values. Which is not true.